### PR TITLE
Makefile: clean golden before re-generating them

### LINKED ISSUE #1165: Improve golden file generation functionality
Right now when we hit `make generated` it generates golden files for test scenarios and also updates the already existent files for test scenarios. But if there is some name change in the test name, it generates a new golden file, and the old one becomes redundant and remains there forever.

This function should be improved to remove all golden files and generate new ones so that we don't end with redundant golden files.

### PR DESCRIPTION
**Pull request by vdemeester**
# Changes

That way, we make sure we clean golden file that shouldn't exists
anymore ; in case of a test rename for example.

Closes #1165 

/cc @piyush-garg @chmouel @pradeepitm12 @danielhelfand 

Signed-off-by: Vincent Demeester <vdemeest@redhat.com>

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [x] Run the code checkers with `make check`
- [x] Regenerate the manpages, docs and go formatting with `make generated`
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/cli/blob/master/CONTRIBUTING.md)
for more details._

# Release Notes

```release-note
NONE
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -89,10 +89,15 @@ test-unit: ./vendor ## run unit tests
...
 .PHONY: update-golden
-update-golden: ./vendor ## run unit tests (updating golden files)
+update-golden: clean-golden ./vendor ## run unit tests (updating golden files)
 	@echo "Running unit tests to update golden files..."
 	@./hack/update-golden.sh
 
+.PHONY: clean-golden
+clean-golden:
+	@echo "Clean golden files..."
+	-find -iname "*.golden" -delete
```
Comment (piyush-garg) [inline Makefile:99]: Just one nit -> exclude vendor, right now if don't have any file but may be in furture

Comment (vdemeester) [reply Makefile:99]: good point

**Comment by chmouel** [comment]
/lgtm
/approve

Thanks 👍🏼🔥

**Comment by vdemeester** [comment]
Hum… so some e2e are using the golden feature.. and `./hack/update-golden.sh` doesn't take those into account :thinking: :sweat_smile:

**Pull request closed without merging**